ui/admin/app/components/model/UpdateModelUsage.tsx

```diff
@@ -0,0 +1,49 @@
+import { memo, useState } from "react";
+
+import { Model, ModelUsage, getModelUsageLabel } from "~/lib/model/models";
+import { ModelApiService } from "~/lib/service/api/modelApiService";
+
+import {
+    Select,
+    SelectContent,
+    SelectItem,
+    SelectTrigger,
+    SelectValue,
+} from "~/components/ui/select";
+
+export const UpdateModelUsage = memo(function UpdateModelUsage({
+    model,
+    onChange,
+}: {
+    model: Model;
+    onChange?: (usage: ModelUsage) => void;
+}) {
+    const [usage, setUsage] = useState(model.usage);
+    const handleModelUsageChange = (value: string) => {
+        const updatedUsage = value as ModelUsage;
+        ModelApiService.updateModel(model.id, {
+            ...model,
+            usage: updatedUsage,
+        });
+        setUsage(updatedUsage);
+        onChange?.(updatedUsage);
+    };
+
+    return (
+        <Select onValueChange={handleModelUsageChange} value={usage}>
+            <SelectTrigger>
+                <SelectValue placeholder="Select Usage..." />
+            </SelectTrigger>
+
+            <SelectContent>
+                {Object.entries(ModelUsage).map(([key, value]) =>
```

Comment:
Looking into why when this input is added, I get the flickering rerendering happening (but doesn't happen at all when this is not there.